tools/mo/openvino/tools/mo/middle/passes/infer.py

```diff
@@ -220,10 +220,33 @@ def override_batch(graph: Graph, batch: int):
     batch: user defined integer value to override batch
     """
     if batch is not None:
-        for node_id, data in graph.nodes(data=True):
-            if 'op' in data and data['op'] == 'Parameter' and not data.get('fixed_batch', False):
-                validate_batch_in_shape(data['shape'], data['name'])
-                data['shape'][0] = batch
+        in_nodes = graph.get_op_nodes(op='Parameter')
+        layout_values = None
+        if 'layout_values' in graph.graph['cmd_params']:
+            layout_values = graph.graph['cmd_params'].layout_values.copy()
```

Comment:
minor: the same code in 2 places, for me looks better if move it to separate function